### PR TITLE
Be able to define time window rules for prefill queries in templates

### PR DESCRIPTION
**Pull request by mkuzdowicz**
## What's changed?

Be able to define time window rules for prefill queries in templates

## Implementation notes

Time window config is specified on Edition Template level

for example lets say for DailyEdition it will be:

```
capiQueryPrefillParams = CapiQueryPrefillParams(
      timeWindowConfig = TimeWindowConfigInDays(
        startDiff = -1,
        endDiff = 3
      )
    )
```

then it means that for issue-date is `2019-OCT-10` for window (-1, 3) the fromDate will be `2019-OCT-9` toDate will be `2019-OCT-13`

those values are propagated to Capi query params:

`to-date=2019-10-07T00%3A00%3A00Z`
`from-date=2019-10-04T00%3A00%3A00Z`

Now all templates have `startDiff = 0` and `endDiff = 0 `so it remain the same as before in runtime

Next steps would be persisting those window settings in DB

And having a new PathType such that those window rules will apply to  specified path type and to not disturb crosswords

### General
- [x] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [x] 🔍 Checked on CODE

### CODE REVIEW

File: app/model/editions/EditionsTemplates.scala [hunk 1]
```diff
@@ -146,8 +146,12 @@ case class FrontTemplate(
   def swatch(swatch: Swatch) = copy(presentation = FrontPresentation(swatch))
 }
 
+case class TimeWindowConfigInDays(startDiff: Int, endDiff: Int)
+case class CapiQueryPrefillParams(timeWindowConfig: TimeWindowConfigInDays)
```
Comment (akash1810) [inline app/model/editions/EditionsTemplates.scala]: Could use a default here to have a simpler interface:

```suggestion
case class CapiQueryPrefillParams(timeWindowConfig: TimeWindowConfigInDays = TimeWindowConfigInDays(0, 0))
```

File: app/services/editions/prefills/PrefillHelper.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package services.editions.prefills
```
Comment (akash1810) [inline app/services/editions/prefills/PrefillHelper.scala]: What's the reason for this change? That is, why are we extracting `geneneratePrefillQuery` out from `Capi.scala`?

Comment (mkuzdowicz) [reply app/services/editions/prefills/PrefillHelper.scala]: I wanted to have a package related to prefills solely

File: app/model/editions/EditionsTemplates.scala (hunk 1 above)
Comment (mkuzdowicz) [reply app/model/editions/EditionsTemplates.scala]: I had that at the beginning, but then i decided that it is more clear to specify those values in templets config so it will be clear which values are used by which edition
Default values can cause that we may forget that those configs exists

File: app/model/editions/EditionsTemplates.scala [hunk 3]
```diff
@@ -146,8 +146,12 @@ case class FrontTemplate(
   def swatch(swatch: Swatch) = copy(presentation = FrontPresentation(swatch))
 }
 
+case class TimeWindowConfigInDays(startDiff: Int, endDiff: Int)
```
[the remark this answers is not in the document]
Comment (mkuzdowicz) [reply app/model/editions/EditionsTemplates.scala]: `startOffset` and `endOffset` ?

File: test/editions/EditionTemplates.scala [hunk 4]
```diff
@@ -15,58 +13,61 @@ class editionTemplateTest extends FreeSpec with Matchers {
...
-    override def getPrefillArticleItems(issueDate: LocalDate, capiPrefillQuery: CapiPrefillQuery): Future[List[Prefill]] = Future.successful(Nil)
+
+    override def getPrefillArticles(prefillParams: PrefillParamsAdapter, currentPageCodes: List[String]): Future[SearchResponse] = Future.successful(null)
+
+    override def getPrefillArticleItems(prefillParams: PrefillParamsAdapter): Future[List[Prefill]] = Future.successful(Nil)
   }
+
   val templating = new EditionsTemplating(TestEdition.templates, TestCapi)
 
-//  "createEdition" - {
```
Comment (akash1810) [inline test/editions/EditionTemplates.scala]: Not for this PR, but this is of no value commented out. We should either fix the test, or delete this.

Comment (mkuzdowicz) [reply test/editions/EditionTemplates.scala]: agree, i was planning to do that later anyway
